src/calendar.cpp

```diff
@@ -541,9 +542,15 @@ time_duration calendar::year_length()
 
 time_duration calendar::season_length()
 {
-    static const std::string s = "SEASON_LENGTH";
-    // Avoid returning 0 as this value is used in division and expected to be non-zero.
-    return time_duration::from_days( std::max( get_option<int>( s ), 1 ) );
+    return time_duration::from_days( std::max( cur_season_length, 1 ) );
```

Comment:
Would it make more sense to do the `std::max` and conversion to a `time_duration` inside `set_season_length`?

Reply:
Yes, it should! No reason to repeatedly do a max operation.